korgw-sample/src/commonMain/kotlin/com/soywiz/korgw/sample/KorgwSample.kt

```diff
@@ -68,144 +53,103 @@ class KCube {
+    var projMat = Matrix3D().setToPerspective(aov, ar, near, far)
+    var viewMat = Matrix3D().setToTranslation(xPos, yPos, zPos)
+    var modlMat = Matrix3D().identity()
+
+    fun run() = Korio {
+        DefaultGameWindow.loop {
+            //TODO: why can't I use width and height vars here ?
+            configure(400, 400, "Multicolored cube", fullscreen = false)
+            addEventListener<ReshapeEvent> {
+                reshape(it, ag)
+            }
             addEventListener<RenderEvent> {
                 render(ag)
             }
         }
-
-    fun init(ag:AG) {
-        //val gl = (ag as AGAwt).gl ??
-        //gl.clearColor(0f,0f,1f,1f)
-        //gl.clearDepthf(1.0f)
-        //gl.enable(gl.DEPTH_TEST)
-        // gl.depthFunc(gl.LEQUAL)
     }
 
-    private var vertices: AG.Buffer? = null
-    private var verticesData: FloatArrayList? = null
+    fun reshape(event: ReshapeEvent, ag: AG) {
+        width = event.width
+        height = event.height
+        if (height <= 0)
+            height = 1
+        ar = (width.toFloat() / height.toFloat())
+        ag.setViewport(0, 0, width, height)
+        projMat = Matrix3D().setToPerspective(aov, ar, near, far)
+    }
 
-    fun render(ag:AG) {
-        if (!init) {
-            init(ag)
-            init = true
+    fun render(ag: AG) {
+        ag.clear(color = Colors.BLACK, depth = far)
+
+        modlMat = Matrix3D().identity().setToRotation(rquad, rotAxis);
+
+        val points = mutableListOf<Float>()
+        points += drawCube()
+        ag.createVertexBuffer(points.toFloatArray()).use { vertices ->
+            ag.draw(
+                vertices,
+                program = prog,
+                type = AG.DrawType.TRIANGLES,
+                vertexLayout = VertexLayout(point, a_col),
+                vertexCount = points.size / 6,
+                uniforms = AG.UniformValues(
+                    u_ProjMat to projMat,
+                    u_ViewMat to viewMat,
+                    u_ModMat to modlMat
+                ),
+                renderState = rs
+            )
         }
-        //ag.clear(color = Colors.ALICEBLUE, depth = 0f)
-        ag.clear(color = Colors.ALICEBLUE, depth = 1f)
-
-        if (vertices == null) {
-            vertices = ag.createVertexBuffer()
-
-            // Updating
-            verticesData = FloatArrayList()
-
-            fun vertex(x: Float, y: Float, z: Float, color: RGBA) {
-                verticesData!!.add(floatArrayOf(x, y, z, color.rf, color.gf, color.bf))
-            }
-
-            fun quad(x: Float, y: Float, width: Float, height: Float, color: RGBA, z: Float) {
-                val x0 = x
-                val x1 = x + width
-                val y0 = y
-                val y1 = y + height
 
-                vertex(x0, y0, z, color)
-                vertex(x1, y0, z, color)
-                vertex(x0, y1, z, color)
+        rquad = rquad.plus(Angle.fromDegrees(0.15))
```

Comment:
`rquad += 0.15.degrees` should do

Reply:
ok, however, FYI, minus does not work.

Reply:
https://github.com/korlibs/korma/commit/996c26a0f8b42d35edcf96c8eff812584875ca3e